TerminalDocs/command-palette.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Windows Terminal Command Palette
+description: Learn how to use the command palette in the Windows Terminal.
+author: cinnamon-msft
+ms.author: cinnamon
+ms.date: 08/26/2020
+ms.topic: how-to 
+ms.service: terminal
+---
+
+# How to use the command palette in Windows Terminal ([Preview](https://aka.ms/terminal-preview/))
+
+The command palette lets you see which commands you can run inside Windows Terminal.
+
+## Invoking the command palette
+
+You can invoke the command palette by typing `ctrl+shift+p`. This can be customized by adding the `commandPalette` command to your key bindings.
+
+```json
+{ "command": "commandPalette", "keys": "ctrl+shift+p" }
+```
+
+## Command line mode
+
+If you'd like to enter a `wt` command into the command palette, you can do so by starting with the `>` character. This will run the `wt` command in the current window.
+
+## Defining commands
+
+There are three different ways you can define a command in Windows Terminal.
+
+1. With a value for `keys` without a `name`
+
+This is the behavior that is documented on the [key bindings page](./customize-settings/key-bindings.md). These commands can be invoked with the defined key binding. The commands will appear in the command palette if Windows Terminal can autogenerate a reasonable name for it.
+
+```json
+{ "command": "newTab", "keys": "ctrl+shift+t" }
+```
+
+2. Without a value for `keys`, but with a `name`
+
+These commands will only be added to the command palette. They cannot be invoked with a key binding. `name` overrides the autogenerated name.
+
+```json
+{ "name": "Open my default tab", "action": "newTab" }
+```
+
+3. With a value for `keys` with a `name`
+
+These commands will be added to the command palette and can be invoked with the keyboard. The `name` property will override the autogenerated name.
+
+```json
+{ "name": "New tab", "command": "newTab", "keys": "ctrl+shift+t" }
+```
+
+### Adding an icon
+
+You can optionally add an icon to a command that appears in the command palette. This can be done by adding the `icon` property to the command.
+
+```json
+{ "icon": "", "name": "New tab", "command": "newTab", "keys": "ctrl+shift+t" }
+```
+
+### Nested commands
+
+Nested commands let you invoke multiple commands with just one command. These will appear as single items in the command palette and can be invoked with one key binding. This can be done with the following syntax:
+
+### Iterable commands
+
+Iterable commands let you iterate over an array defined in your settings.json file. Possible arrays to iterate over include `profiles` and `schemes`. These will create individual items in the command palette, one for each iteration. Below is an example of iterating over each profile and creating a new tab for each one.
```

Comment:
The more context the better here. I'm thinking something like:

<hr>

Iterable commands let you create multiple commands at the same time, generated from other objects defined in your settings. Currently, you can create iterable commands for your profiles and color schemes. At runtime, these commands will be expanded to one command for each of the objects of the given type. For example:

\<gif>
\<code>

(then, below the break)

In the above example:
* `"iterateOn": "profiles"` told the Terminal "I'd like one of these commands for each of my profiles"
* At runtime, the Terminal will replace `${profile.icon}` with whatever the path to a profile's icon is, and `${profile.name}` with whatever the profile's name is.

So if you had three profiles:
```jsonc
"profiles": [
	{ "name": "CMD", "icon": null },
	{ "name": "PowerShell", "icon": "C:\\path\\to\\icon.png" },
	{ "name": "Ubuntu", "icon": null },
]
```

then the above command would appear as the following three commands:
```json
{
    "name": "New tab...",
    "commands": [
        {
            "icon": null,
            "name": "CMD",
            "command": { "action": "newTab", "profile": "CMD" }
        },
        {
            "icon": "C:\\path\\to\\icon",
            "name": "PowerShell",
            "command": { "action": "newTab", "profile": "PowerShell" }
        },
        {
            "icon": null,
            "name": "Ubuntu",
            "command": { "action": "newTab", "profile": "Ubuntu" }
        }
    ]
},
```

Reply:
Then we'd probably need to also list that for commands iterable on `profiles`, they can use `${profile.name}` and `${profile.icon}`, and for `schemes`, they can use `${scheme.name}`.

We might want to include the color scheme example in the defaults.json (or a link to it) as a reference even.